components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_account/select_account.tsx

```diff
@@ -0,0 +1,216 @@
+
+// Components
+import {
+  withPlaceholderIcon //
+} from '../../../../../components/shared/create-placeholder-icon/index'
+import { NftIcon } from '../../../../../components/shared/nft-icon/nft-icon'
+import {
+  CreateNetworkIcon //
+} from '../../../../../components/shared/create-network-icon'
+import {
+  CreateAccountIcon //
+} from '../../../../../components/shared/create-account-icon/create-account-icon'
+
+// Styles
+import {
+  Column,
+  ScrollableColumn,
+  LeoSquaredButton,
+  Row,
+  Text
+} from '../../../../../components/shared/style'
+import {
+  IconsWrapper,
+  AssetIcon,
+  NetworkIconWrapper,
+  AccountButton
+} from './select_account.style'
+
+interface Props {
+  token: BraveWallet.BlockchainToken
+  accounts: BraveWallet.AccountInfo[]
+  tokenBalancesRegistry?: TokenBalancesRegistry | null
+  spotPrice?: BraveWallet.AssetPrice
+  onSelectAccount: (account: BraveWallet.AccountInfo) => void
+}
+
+const ICON_CONFIG = { size: 'big', marginLeft: 0, marginRight: 0 } as const
+const AssetIconWithPlaceholder = withPlaceholderIcon(AssetIcon, ICON_CONFIG)
+const NftIconWithPlaceholder = withPlaceholderIcon(NftIcon, ICON_CONFIG)
+
+export const SelectAccount = (props: Props) => {
+  const { token, accounts, tokenBalancesRegistry, spotPrice, onSelectAccount } =
+    props
+
+  // Queries
+  const { data: tokensNetwork } = useGetNetworkQuery(token)
+  const { data: defaultFiatCurrency = 'usd' } = useGetDefaultFiatCurrencyQuery()
+
+  // State
+  const [selectedAccount, setSelectedAccount] = React.useState<
+    BraveWallet.AccountInfo | undefined
+  >(accounts[0] || undefined)
+
+  // Methods
+  const getTokenBalance = React.useCallback(
+    (account: BraveWallet.AccountInfo) => {
+      return getBalance(account.accountId, token, tokenBalancesRegistry)
+    },
+    [tokenBalancesRegistry, token]
+  )
+
+  const getFiatBalance = React.useCallback(
+    (account: BraveWallet.AccountInfo) => {
+      return spotPrice
+        ? new Amount(getTokenBalance(account))
+            .divideByDecimals(token.decimals)
+            .times(spotPrice.price)
+            .formatAsFiat(defaultFiatCurrency)
+        : Amount.empty().formatAsFiat(defaultFiatCurrency)
+    },
+    [getTokenBalance, spotPrice, token, defaultFiatCurrency]
+  )
+
+  const onContinue = React.useCallback(() => {
+    if (selectedAccount) {
+      onSelectAccount(selectedAccount)
+    }
+  }, [onSelectAccount, selectedAccount])
+
+  return (
+    <Column
+      fullWidth={true}
+      fullHeight={true}
+      justifyContent='flex-start'
+    >
+      <Column margin='0px 0px 24px 0px'>
+        <IconsWrapper margin='0px 0px 8px 0px'>
+          {token.isErc721 || token.isNft ? (
+            <NftIconWithPlaceholder
+              asset={token}
+              network={tokensNetwork}
+            />
+          ) : (
+            <AssetIconWithPlaceholder
+              asset={token}
+              network={tokensNetwork}
```

Comment:
VSCode highlights me that network prop is deprecated

Reply:
Yes, I saw that as well. Is currently required as a prop for some reason... is probably best we remove the prop in a separate PR, since `withPlaceholderIcon` is used in a lot of places throughout the `Wallet`.

Reply:
You could change the prop to optional for now

Reply:
Good idea @josheleonard, made `network` optional!

Reply:
Created an issue to remove https://github.com/brave/brave-browser/issues/38217